external/avro/src/main/scala/org/apache/spark/sql/avro/AvroOptions.scala

```diff
@@ -47,7 +51,21 @@ private[sql] class AvroOptions(
    * schema converted by Spark. For example, the expected schema of one column is of "enum" type,
    * instead of "string" type in the default converted schema.
    */
-  val schema: Option[String] = parameters.get("avroSchema")
+  val schema: Option[Schema] = {
+
+    val avroUrlSchema = parameters.get("avroSchemaUrl").map(schemaFSUrl => {
+      log.info("loading avro schema from url: " + schemaFSUrl)
+      val fs = FileSystem.get(new URI(schemaFSUrl), conf)
+      val in = fs.open(new Path(schemaFSUrl))
```

Comment:
I feel like this is just a small code bit that can be just done in user's application side.

Reply:
1. this is the behavior when using Avro Hive table. see `org.apache.hadoop.hive.serde2.avro.AvroSerdeUtils.determineSchemaOrThrowException()`. so now when we're trying to move users from the Avro table to `spark.read.` they have a regression.
2. some of the users use mainly sql code. it will be cumbersome for them to write this logic every time.